### PR TITLE
Adds support for Magic Transit IPsec tunnels

### PR DESCRIPTION
**Pull request by cehrig**
## Description

Adds support for Magic Transit IPsec Tunnels. 

**API docs:**
List Tunnels: https://api.cloudflare.com/#magic-ipsec-tunnels-list-ipsec-tunnels
Get Tunnel: https://api.cloudflare.com/#magic-ipsec-tunnels-ipsec-tunnel-details
Create Tunnel: https://api.cloudflare.com/#magic-ipsec-tunnels-create-ipsec-tunnels
Update Tunnel: https://api.cloudflare.com/#magic-ipsec-tunnels-update-ipsec-tunnel
Delete Tunnel: https://api.cloudflare.com/#magic-ipsec-tunnels-delete-ipsec-tunnel

## Has your change been tested?

``` 
$ go test $(go list ./...) -v -run "^Test.*MagicTransitIPsecTunnel.*$"
=== RUN   TestListMagicTransitIPsecTunnels
--- PASS: TestListMagicTransitIPsecTunnels (0.00s)
=== RUN   TestGetMagicTransitIPsecTunnel
--- PASS: TestGetMagicTransitIPsecTunnel (0.00s)
=== RUN   TestCreateMagicTransitIPsecTunnels
--- PASS: TestCreateMagicTransitIPsecTunnels (0.00s)
=== RUN   TestUpdateMagicTransitIPsecTunnel
--- PASS: TestUpdateMagicTransitIPsecTunnel (0.00s)
=== RUN   TestDeleteMagicTransitIPsecTunnel
--- PASS: TestDeleteMagicTransitIPsecTunnel (0.00s)
PASS
```
Tested using a production account as well. The Create and Update endpoints occasionally run into a time out. I am checking with Magic team why that happens.

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: magic_transit_ipsec_tunnel.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+// ListMagicTransitIPsecTunnels lists all IPsec tunnels for a given account
+//
+// API reference: https://api.cloudflare.com/#magic-ipsec-tunnels-list-ipsec-tunnels
+func (api *API) ListMagicTransitIPsecTunnels(ctx context.Context) ([]MagicTransitIPsecTunnel, error) {
+	if err := api.checkAccountID(); err != nil {
+		return []MagicTransitIPsecTunnel{}, err
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/magic/ipsec_tunnels", api.AccountID)
```
Comment (jacobbednarz) [inline magic_transit_ipsec_tunnel.go:77]: can we refactor this to instead accept the account ID as a parameter? api.AccountID isn’t concurrency safe and is causing issues with some Tenant support.

Comment (cehrig) [reply magic_transit_ipsec_tunnel.go]: Makes sense, thank you for raising. I have changed the signatures.

**Pull request merged**